src/plugins/lexer/mod.ts

```diff
@@ -0,0 +1,390 @@
+    if (
+      (this.token.length === 0) && (this.mode !== Mode.String) &&
+      (this.mode !== Mode.Compiler)
+    ) {
+      this.trace.column[0] = this.virtual_position;
+    }
+    // verify mode lexer is in
+    switch (this.mode) {
+      case Mode.Token: {
+        // switch to string mode
+        if ((character === '"') && (this.token.length === 0)) {
+          this.mode = Mode.String;
+          break;
+
+          // presume token is finished; generate
+        } else if (character === " ") {
+          if (this.token.length === 0) return;
+          this.checkToken();
+
+          // switch to compiler call mode
+        } else if (character === "&") {
+          this.mode = Mode.Compiler;
+          this.checkToken();
+
+          // check if current character is an operator
+        } else if (this.isOperator(character) === true) {
+          // handles operator being beside another token
+          this.checkToken();
+          this.token += character;
+          // makes sure column position is accurate for operator
+          this.trace.column[0] = this.virtual_position;
+          this.checkToken(1);
+        } else {
+          // if no conditions met, keep building token
+          this.token += character;
+        }
+        break;
+      }
+
+      case Mode.String: {
+        // check if end of string was reached; switch back to `token` mode
+        if ((character === '"')) {
+          this.mode = Mode.Token;
+          this.trace.column[1] = this.virtual_position;
+          this.createToken("string", this.token);
+
+          // check if next character should be escaped
+        } else if (character === "\\") {
+          this.token += this.next();
+
+          // if no conditions met, keep building string
+        } else {
+          this.token += character;
+
+          /*
+          if string wasnt closed correctly, auto close and store
+          string with absorbed tokens from line for the parser to
+          use in error output; continue lexing to discover possible
+          errors ahead
+          */
+          if (this.peek() === "&") {
+            this.mode = Mode.Token;
+            this.trace.column[1] = this.virtual_position;
+            this.createToken("invalid_string", this.token);
+          }
+        }
+        break;
+      }
+
+      case Mode.Compiler: {
+        this.token += character;
+
+        /*
+        - basic checking for end-of-line / end-of-file tokens
+        - columns are negative since they don't exist in .mtl file
+
+        TODO:
+        - related with the comment on the function far below,
+          this source could be configured better to work similarly
+          to string / token lexing. works as intended for now.
+        */
+        switch (checkForString(this.token, ["eol", "eof"])) {
+          case "eol": {
+            this.trace.column = [-1, -1];
+            this.createToken("eol", "*");
```

Comment:
What's the rationale for using `*` as a value in eol & eof tokens? Wouldn't a null or empty string value be more appropriate?

Reply:
More so a placeholder value, will clear.